prebid/native-implementation.md

```diff
@@ -191,158 +268,428 @@ pbjs.addAdUnits({
+- Prebid.js responds with OpenRTB response that contains `eventtrackers`
+- When the Prebid Universal Creative renders the native ad, it will gather all impression trackers from the response and for each tracker it will:
+  - if tracker is `img` method, Prebid Universal Creative will fire pixel provided in the `url` property
+  - if tracker is `js` method, Prebid Universal Creative will load the script in the same iframe where native ad is rendered
+
+The publisher doesn't need to implement anything for impression tracking to work, all that is needed is proper configuration of OpenRTB request on adUnit level. Prebid.js + Prebid Universal Creative will automatically take care of the impression tracking.
+
+### 6.2. Click Tracking
+According to the OpenRTB spec, click trackers can be found in `link` objects. There is a 'master' `link` object for the entire native ad and also each asset in the response can have its own separate `link` object.
+
+When a native ad is rendered, the Prebid Universal Creative will attach click listeners on all DOM elements that have class `pb-click`. Which means it's up to the publisher to decide which DOM objects are 'clickable'.
+
+By default, only the 'master' `link` URLs will be fired, but, the Prebid Universal Creative can also fire click tracking for specific assets. To enable this, publisher assigns a custom attribute to the associated DOM element.
+
+```html
+<div class="pb-click" hb_native_asset_id = "5">
+```
+If the user clicks on this div, the Prebid Universal Creative will take the `link` object from the identified asset and fire any click trackers. If the asset doesn't have a `link` object, it will just fire the click trackers associated with 'master' `link` object.
+
+Example of how to configure a template so that the Prebid Universal Creative can fire click trackers:
+``` html
+<div class="sponsored-post">
+  <div class="thumbnail" style="background-image: url(##hb_native_asset_id_1##);"></div>
+  <div class="content">
+    <h1><a href="%%CLICK_URL_UNESC%%##hb_native_linkurl##" target="_blank" class="pb-click" hb_native_asset_id="2">##hb_native_asset_id_2##</a></h1>
+    <p class="pb-click" hb_native_asset_id="4">##hb_native_asset_id_4##</p>
+    <div class="attribution" class="pb-click" hb_native_asset_id="3">##hb_native_asset_id_3##</div>
+  </div>
+</div>
+```
+
+### 6.3. Viewability Tracking
+In order for viewability tracking to work, there needs to be a script on `window.top` that is tracking if the ad's iframe is viewable or not. Also, this script needs to communicate with Prebid.js so that appropriate trackers are fired when the ad is viewable.
+
+To minimize needed work on the publisher side, we recommend using the [Prebid.js Viewability module](https://github.com/prebid/Prebid.js/blob/master/modules/viewability.md).
+
+Enabling viewability tracking for native ads:
+- Build your Prebid.js bundle with the `viewability` module
```

Comment:
This paragraph has been completely removed, because the viewability module has been removed too. This is because the module had a security issue and needs to be fixed; until it happens, it is better not to link it to discurage bad, unsecure practices.